exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -228,17 +230,17 @@ def _translate_to_prometheus(
 
             pre_metric_family_ids = []
 
-            metric_name = ""
-            metric_name += self._sanitize(metric.name)
+            metric_name = sanitize_full_name(metric.name)
```

Comment:
> The resulting unit SHOULD be added to the metric as [UNIT metadata](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metricfamily) and as a suffix to the metric name unless the metric name already contains the unit, or the unit MUST be omitted.

Doesn't the name need to have the unit appended as a suffix if it doesn't exist already in the name?